### PR TITLE
Track UI translations

### LINKED ISSUE #1058: Track UI translation progress in the Translation Status Overview
We are currently only tracking changes in the Markdown pages inside the Translation Status Overview Issue, it would be nice if we also tracked the changes in the `i18n/` folder (`nav.ts`, `ui.ts`) and in the new `public/captions` folder, so translators are aware of the changes made to these files in the English docs. 

**Note:** We also have to reflect these changes in the i18n guide so translators are aware that they can translate captions.

### PR DESCRIPTION
**Pull request by TheOtterlord**
#### What kind of changes does this PR include?

- Something else!

#### Description

- Closes #1058 
- What does this PR change? Track changes to UI translations & make their status available via https://i18n.docs.astro.build

As I began PRing this, I noticed a few flaws, so I'm going to make this a draft PR while I explore the alternative. The way this currently works uses the same technique as all other pages (but counts any changes, not just major changes). It can lead to incomplete translations still being marked as complete.

The next stage of this would be to dynamically import all translations, and check that all keys are available before marking it as complete. If we wanted, we could then check to make sure no other changes have been made (if that's something that is likely to make those translations outdated?)

### CODE REVIEW

**Comment by TheOtterlord** [comment]
Updated to first check for key completeness & added missing `isFallback` value to nav translations.

File: src/i18n/translation-checkers.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ export const NavDictionary = (dict: Partial<Record<NavDictionaryKeys, string>>)
 	const orderedDictionary: NavDict = [];
 	for (const enEntry of enNav) {
 		const text = dict[enEntry.key] || enEntry.text;
-		orderedDictionary.push({ ...enEntry, text });
+		orderedDictionary.push({ ...enEntry, text, isFallback: !dict[enEntry.key] });
```
Comment (delucis) [inline src/i18n/translation-checkers.ts:30]: Why was this necessary exactly? Usually our `isFallback` flag does not flag that the _label_ is a fallback, but that the page contents itself are (we use it to display an “EN” suffix in the sidebar for such pages even if the label is translated).

**Review by delucis — COMMENTED** [review]
Thanks for this @TheOtterlord! Exciting we can finally do this. I just have one question.

File: src/i18n/translation-checkers.ts (hunk 1 above)
Comment (TheOtterlord) [reply src/i18n/translation-checkers.ts:30]: Ahh, that explains things. Here, I used this to detect if the label was falling back to english, to check if all labels were provided before marking it as "complete". I'll move this out to a separate value tomorrow

**Comment by TheOtterlord** [comment]
It looks like both the `es` and `it` translations are not using `NavDictionary`? Looks like this could be causing the View Transitions page to not appear in the sidebar of the Italian translation.
It's a separate issue, which I'll open now. But since this PR assumes that all languages use `NavDictionary`, and flags them as out of date if they don't, should we hold off merging for now?

**Comment by Yan-Thomas** [comment]
Just merged the PR you mentioned @TheOtterlord, can you take a look to see if what you expected is working? I'll review right after!

**Comment by TheOtterlord** [comment]
Yep, confirmed that everything works correctly now

**Review by Yan-Thomas — COMMENTED** [review]
I am seeing there's a few "source change history" (e.g `zh-cn` docsearch & nav.ts) without changes since the final date marked, my guess is that this is related to all changes being counted, not only major changes — is there any reason for that being the case?

**Comment by TheOtterlord** [comment]
Nope, I don't believe so. I'll check this quickly and update it to use the last major commit

**Comment by TheOtterlord** [comment]
Ok, looks like it's not that, but because of the reason it's classed as outdated. If it doesn't have all the keys present, it shows up as outdated. I'll see if I can hide that, and replace it with something more contextual

**Comment by TheOtterlord** [comment]
@Yan-Thomas updated to only display the original, and the translation

updated output for translation dashboard

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Thanks @TheOtterlord, feel free to merge when you can, I have a small comment but it isn't blocking!

File: scripts/lib/translation-status/builder.ts [hunk 2]
```diff
@@ -14,6 +14,12 @@ import type {
 	PageTranslationStatus,
 } from '../../lib/translation-status/types';
 import { toUtcString, tryGetFrontMatterBlock } from '../../lib/translation-status/utils.js';
+import docsearchTranslations from '../../../src/i18n/en/docsearch'
+import navTranslations from '../../../src/i18n/en/nav'
+import uiTranslations from '../../../src/i18n/en/ui'
+import type { DocSearchTranslation, UIDict } from '~/i18n/translation-checkers';
```
[the remark this answers is not in the document]
Comment (TheOtterlord) [reply scripts/lib/translation-status/builder.ts:20]: The aliases aren't correctly resolved at runtime

[the remark this answers is not in the document]
Comment (TheOtterlord) [reply scripts/lib/translation-status/builder.ts:20]: looks to be an open `tsm` issue: https://github.com/lukeed/tsm/issues/6
